clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -307,6 +307,26 @@ def CIR_ArrayType : CIR_Type<"Array", "array",
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// VectorType (fixed size)
+//===----------------------------------------------------------------------===//
+
+def CIR_VectorType : CIR_Type<"Vector", "vector",
+    [DeclareTypeInterfaceMethods<DataLayoutTypeInterface>]> {
+
+  let summary = "CIR vector type";
+  let description = [{
+    `cir.vector' represents fixed-size vector types.  The parameters are the
+    element type and the number of elements.
+  }];
+
+  let parameters = (ins "mlir::Type":$eltType, "uint64_t":$size);
```

Comment:
CIR uses mix of `eltType`, `elementTy`. Can we standardize one? tbf, I would prefer third option `elementType` to mirror similar options in other dialects (llvm, builtin, tosa)

Reply:
We can agree on which one will be standardised and i will use it in this PR, and in the follow up PR I can update the other places to use that one

@xlauko @bcardosolopes @andykaylor

Reply:
Since @xlauko is doing a lot of work on making things nice and consistency, I'm all in for his suggestions. Also probably worth encoding in https://llvm.github.io/clangir/Dialect/cir-style-guide.html so we can point people to it.